webapp/src/Controller/API/JudgehostController.php

```diff
@@ -368,30 +368,41 @@ public function updateJudgingAction(Request $request, string $hostname, int $jud
             $output_compile = base64_decode($request->request->get('output_compile'));
             if ($request->request->getBoolean('compile_success')) {
                 if ($judging->getOutputCompile() === null) {
-                    $judging
-                        ->setOutputCompile($output_compile)
-                        ->setJudgehost($judgehost);
+                    $judging->setOutputCompile($output_compile);
                     $this->em->flush();
 
                     $this->eventLogService->log('judging', $judging->getJudgingid(),
                         EventLogService::ACTION_CREATE, $judging->getContest()->getCid());
                 } elseif ($judging->getResult() === Judging::RESULT_COMPILER_ERROR) {
                     // The new result contradicts a former one, that's not good.
-                    // Since the other judgehost was not successful, but we were , assume that the other judgehost
-                    // is broken and disable it.
-                    $disabled = [
-                        'kind' => 'judgehost',
-                        'hostname' => $judging->getJudgehost()->getHostname(),
-                    ];
-                    $error = new InternalError();
-                    $error
-                        ->setJudging($judging)
-                        ->setContest($judging->getContest())
-                        ->setDescription('Compilation results are different for j' . $judging->getJudgingid())
-                        ->setJudgehostlog(base64_encode('New compilation output: ' . $output_compile))
-                        ->setTime(Utils::now())
-                        ->setDisabled($disabled);
-                    $this->em->persist($error);
+                    // Since the other judgehosts were not successful, but we were , assume that the other judgehosts
+                    // are broken and disable it.
+                    $disableHostnames = [];
+                    /** @var JudgingRun $run */
+                    foreach ($judging->getRuns() as $run) {
+                        if ($run->getJudgeTask() &&
+                            $run->getJudgeTask()->getJudgehost() &&
+                            $run->getJudgeTask()->getJudgehost()->getHostname() !== $judgehost->getHostname()) {
+                            $hostname = $run->getJudgeTask()->getJudgehost()->getHostname();
+                            $disableHostnames[$hostname] = $hostname;
```

Comment:
Nit: `disableHostnames[] = $hostname` should do it as well

Reply:
It will be in there multiple times and then we create multiple internal errors. Or we need `array_unique` below

Reply:
I need more ☕